### ISSUE TITLE
i0 changes the shape

### ISSUE THREAD

**Issue by toslunar** [numpy/numpy#11205]
`numpy.i0` squeezes the input, as in an example of the document:
```
>>> np.i0([0.])
array(1.0)
```

This is different from `scipy.special`:
```
>>> special.i0([0.])
array([1.])
```

**Comment by seberg** [comment]
Hmm, well, I somewhat think the easiest thing is probably to just deprecate the function completely asking to use the scipy implementation, it will much better anyway I am sure. Ours is not even a ufunc....

**Comment by seberg** [comment]
A quick code search showed usage mostly in some book or tutorial snipplet(s), so I think we should get away with direct deprecation? Not like it needs to be rushed, can go very slowly, deprecated -> visible deprecate, etc. And mostly add a note to the decumentation. It is python code, so should be a simple thing for someone wanting to do something small.
Probably needs a ping on the mailing list, just to be sure nobody feels strongly about fixing it instead of just deprecating.

**Comment by mhvk** [comment]
@seberg - `i0` is also used inside `kaiser` - but I guess you mean to make `i0` private? It is not a difficult fix...

**Comment by seberg** [comment]
Ah I missed that it is used there, I guess for the kaiser the private version may be good enough, or we should just try to clean it up, it is also not that bad, just thought it might be cleaner anyway. I would have searched for the bessel functions in scipy.special to begin with...

**Comment by mhvk** [comment]
Fix in #11233.
